src/riverroute/RtmRestFile.F90

```diff
@@ -235,39 +234,47 @@ end subroutine RtmRestGetfile
 !-----------------------------------------------------------------------
 
   subroutine restFile_read_pfile( pnamer )
-
     ! !DESCRIPTION:
     ! Setup restart file and perform necessary consistency checks
+    !!USES:
+    use mpi, only : MPI_CHARACTER
+    use RtmSpmd, only : mpicom_rof
 
     ! !ARGUMENTS:
     implicit none
     character(len=*), intent(out) :: pnamer ! full path of restart file
 
     ! !LOCAL VARIABLES:
-    integer :: i                  ! indices
-    integer :: nio                ! restart unit
-    integer :: status             ! substring check status
-    character(len=256) :: locfn   ! Restart pointer file name
-    !--------------------------------------------------------
-
-    ! Obtain the restart file from the restart pointer file.
-    ! For restart runs, the restart pointer file contains the full pathname
-    ! of the restart file. For branch runs, the namelist variable
-    ! [nrevsn_rtm] contains the full pathname of the restart file.
-    ! New history files are always created for branch runs.
+    integer :: nio                 ! restart unit
+    integer :: ier                 ! error return from fortran open
+    integer :: i                   ! indices
+    integer :: yr, mon, day, tod   ! Year, month, day of month, and time-of-day
+    character(len=17) :: timestamp ! Simulation timestamp for current date
+    character(len=256) :: locfn    ! Restart pointer file name
+    !-------------------------------------
 
     if (masterproc) then
-       write(iulog,*) 'Reading restart pointer file....'
+       call get_curr_date(yr, mon, day, tod)
+       write(timestamp,'(".",i4.4,"-",i2.2,"-",i2.2,"-",i5.5)'),yr,mon,day,tod
+       locfn = './'// trim(rpntfil)//trim(inst_suffix)//timestamp
+
+       write(iulog,*) 'Reading restart pointer file: '//trim(locfn)
+       open (newunit=nio, file=trim(locfn), status='old', form='formatted', iostat=ier)
```

Comment:
@slevis-lmwg note here that I went away from using the RTM subroutine opnfil, partially because the new FORTRAN syntax for newunit means it's not as needed anymore. This maps to what we did in MOSART as well. But, happy to hear an opinion on this change. Going away from use of a central opnfil subroutine in this context and in general is a degrade to modularity.

Reply:
Taking with @slevis-lmwg this is fine so resolving.